scripts/recipes/librispeech/run.sh

```diff
@@ -0,0 +1,97 @@
+# bazel build --cxxopt='-std=c++14' lingvo:trainer third_party/kaldi/... @openfst//... galvasr2/...
+
+if [ $stage -le 1 ]; then
+   local/download_lm.sh $lm_url $work_dir/data/local/lm
+fi
+
+langdir=$work_dir/data/lang_char_1gram
+if [ $stage -le 2 ]; then
+
+    
+    # local/prepare_dict_ctc.sh $work_dir/data/local/lm $work_dir/data/local/dict_phn
+    # steps/ctc_compile_dict_token.sh $work_dir/data/local/dict_phn $work_dir/data/local/lang_phn_tmp $work_dir/data/lang_phn
+    mkdir -p $work_dir/data/local/dict_char
+    # Search for the destination string in librispeech_ctc.py
+    gsutil cp local/tokens.txt gs://the-peoples-speech-west-europe/Librispeech/tokens.txt
+    galvasr_tokenize_words --in_words_txt $work_dir/data/local/lm/librispeech-vocab.txt \
+                           --in_units_txt local/tokens.txt \
+                           --out_spelling_txt $work_dir/data/local/dict_char/lexicon.txt \
+                           --out_spelling_numbers_txt $work_dir/data/local/dict_char/lexicon_numbers.txt \
+                           --out_units_txt $work_dir/data/local/dict_char/units.txt \
+                           --space_char "<space>"
+    steps/ctc_compile_dict_token.sh --dict-type char \
+                                    --space-char "<space>" \
+                                    $work_dir/data/local/dict_char \
+                                    $work_dir/data/local/lang_char_tmp $work_dir/data/lang_char
+    # fstdraw --isymbols=$work_dir/data/lang_char/tokens.txt --osymbols=$work_dir/data/lang_char/tokens.txt $work_dir/data/lang_char/T.fst | dot -Tpdf > T.pdf
+    # fstdraw --isymbols=$work_dir/data/lang_char/tokens.txt --osymbols=$work_dir/data/lang_char/words.txt $work_dir/data/lang_char/L.fst | dot -Tpdf > L.pdf
+    local/format_lms.sh --src-dir $work_dir/data/lang_char $work_dir/data/local/lm
+
+    fsttablecompose ${langdir}/L.fst $langdir/G.fst | fstdeterminizestar --use-log=true | \
+        fstminimizeencoded | fstarcsort --sort_type=ilabel > $langdir/LG.fst || exit 1;
+    # Why isn't this determinized and then minimized? Unclear.
+    fsttablecompose ${langdir}/T.fst $langdir/LG.fst > $langdir/TLG.fst || exit 1;
+    fstconvert --fst_type=const $langdir/TLG.fst $langdir/TLG_const.fst
+
+    # fstarcsort --sort_type=olabel  ${langdir}/T.fst > ${langdir}/T_olabel_sorted.fst
+    # fstarcsort --sort_type=ilabel  ${langdir}/T.fst > ${langdir}/L_ilabel_sorted.fst
+    #     fsttablecompose ${langdir}/T_olabel_sorted.fst $langdir/L_ilabel_sorted.fst | fstdeterminizestar --use-log=true | \
+    #     fstminimizeencoded | fstarcsort --sort_type=ilabel > $langdir/TL.fst || exit 1;
+    # fstconvert --fst_type=const $langdir/TL.fst $langdir/TL_const.fst
+fi
+
+if [ $stage -le 3 ]; then
+    export CUDA_VISIBLE_DEVICES=""
+    export OPENBLAS_NUM_THREADS="1"
+    export MKL_NUM_THREADS="1"
+    # Run training
+    # trainer --logdir=gs://the-peoples-speech-west-europe/training_logs/galvez/tpu_ctc_4m \
```

Comment:
@anjaligopi I uncomment this chunk of code to run the training job. Note that the job type is "trainer_client", not "executor_tpu" (or "tpu_executor", I forgot which it is).

The next call to trainer runs the decoder job asycnhronously with the training job. Basically, it will poll the file system until it finds a new checkpoint has been created, at which point it will start a decode job on the Dev dataset. Note that I am using the "decoder" job type, not the "eval" one for technical reasons having to do with lingvo's weird jargon.